test/unit/index.js

```diff
@@ -104,6 +104,61 @@ test('tracker with minimal token and one block update with no changes', async fu
   }
 })
 
+test('tracker with minimal token and two rapid block updates, the first with changes', async function (t) {
+  t.plan(2)
+  let tracker
+  try {
+    const { addresses, provider, token, tokenAddress } = await setupSimpleTokenEnvironment()
+    tracker = new TokenTracker({
+      pollingInterval: 20,
+      provider,
+      tokens: [
+        {
+          address: tokenAddress,
+        }
+      ],
+      userAddress: addresses[1],
+    })
+
+    const updates = []
+    tracker.on('update', (tokens) => {
+      updates.push(tokens)
+    })
+    await new Promise((resolve) => setTimeout(() => resolve(), 200))
+
+    await Promise.all([
+      token.transfer(addresses[1], '110'),
+      token.transfer(addresses[1], '0'),
+    ])
```

Comment:
Note the simultaneous transfers triggered here. This results in two new blocks (one per transaction, as is the default behavior of testrpc if you don't set a blocktime) in quick succession. This test failed before this change.